src/fetch.js

```diff
@@ -12,20 +12,21 @@ const defaults = {
   }
 };
 
-const handleResponse = response => response
-    .json()
-    .then((json) => {
-      if (response.ok) {
-        return json;
-      }
+const handleResponse = response => Promise
+  .resolve(response)
+  .then(() => (response.status === 204 ? null : response.json()))
```

Comment:
This solves this specific case, but it seems like the right thing to do is look at the content-type and/or content-length headers, which is what [tc-client does](https://github.com/taskcluster/taskcluster-client/blob/fffebc153b5f17dc161127917eaf4c42b1119be3/lib/client.js#L305-L308)

Reply:
tc-client is able to do those things because it's in Node.js and isn't subject to CORS rules around header access. If someone provides a different baseUrl which isn't served with CORS enabling those specific headers, it would still break.

Reply:
Can you fill that in a bit?  Does fetch not allow access to content-type/content-length?

Reply:
Content-Type is allowed (whitelisted), but Content-Length is blacklisted; you cannot access this value from fetch. So I'd say we have 2 options:

- If Content-Type doesn't match `application/json` then skip the `json()` response.
- If status is 204, skip the `json()` response.

Reply:
For what it's worth, I think the content-type check is probably the way to go since it can cover more than 204s.

Reply:
That makes sense.  Thanks for the background!  Blacklisting content-length is weird, but OK!